### ISSUE TITLE
Timezone In Calendar Event Source Only Works on First Payload

### ISSUE THREAD

**Issue by devstein** [argoproj/argo-events#314]
**Describe the bug**
The `timezone` field in the calendar event source only works on the first event

**To Reproduce**
The below configuration first produces an event at `11:45pm PST`, then all proceeding events will be at triggered at  `11:45pm UTC (4:45pm PST)`.

```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: calendar-event-source
  labels:
    # do not remove
    argo-events-event-source-version: v0.10
data:
  end-of-weekdays: |-
    # Argo events chron starts with minutes!
    schedule: "45 23 ? * MON-FRI"
    # userPayload is a static string that will be sent
    # to the sensor with each event payload
    # whatever you put here is blindly delivered to sensor.
    userPayload: "night"
    timezone: "America/Los_Angeles"
---
apiVersion: argoproj.io/v1alpha1
kind: Gateway
metadata:
  name: calendar-gateway
  labels:
    # gateway controller with instanceId "argo-events" will process this gateway
    gateways.argoproj.io/gateway-controller-instanceid: argo-events
    # gateway controller will use this label to match with it's own version
    # do not remove
    argo-events-gateway-version: v0.10
spec:
  type: "calendar"
  eventSource: "calendar-event-source"
  processorPort: "9330"
  eventProtocol:
    type: "HTTP"
    http:
      port: "9300"
  template:
    metadata:
      name: "calendar-gateway"
      labels:
        gateway-name: "calendar-gateway"
    spec:
      containers:
        - name: "gateway-client"
          image: "argoproj/gateway-client:v0.9.3"
          imagePullPolicy: "Always"
          command: ["/bin/gateway-client"]
        - name: "calendar-events"
          image: "argoproj/calendar-gateway:v0.9.3"
          imagePullPolicy: "Always"
          command: ["/bin/calendar-gateway"]
      serviceAccountName: "argo-events-sa"
  watchers:
    sensors:
      - name: "calendar-sensor"
---
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: calendar-sensor
  labels:
    # sensor controller with instanceId "argo-events" will process this sensor
    sensors.argoproj.io/sensor-controller-instanceid: argo-events
    # sensor controller will use this label to match with it's own version
    # do not remove
    argo-events-sensor-version: v0.10
spec:
  template:
    spec:
      containers:
        - name: "sensor"
          image: "argoproj/sensor:v0.9.3"
          imagePullPolicy: Always
      serviceAccountName: argo-events-sa
  dependencies:
    - name: "calendar-gateway:*"
  eventProtocol:
    type: "HTTP"
    http:
      port: "9300"
  triggers:
    - template:
        name: calendar-workflow-trigger
        group: argoproj.io
        version: v1alpha1
        kind: Workflow
        source:
          inline: |
            apiVersion: argoproj.io/v1alpha1
            kind: Workflow
            metadata:
              generateName: calendar-workflow-
            spec:
              entrypoint: whalesay
              arguments:
                parameters:
                - name: message
                  # this is the value that should be overridden
                  value: hello world
              templates:
              - name: whalesay
                inputs:
                  parameters:
                  - name: message
                container:
                  image: docker/whalesay:latest
                  command: [cowsay]
                  args: ["{{inputs.parameters.message}}"]
      resourceParameters:
        - src:
            event: "calendar-gateway:*"
          dest: spec.arguments.parameters.0.value
---
```

**Expected behavior**
The event is always on the `America/Los_Angeles` timezone.

**Environment (please complete the following information):**
 - OS: Linux 
 - Browser [e.g. chrome, safari] `N/A`
 - Version [e.g. 22]: `v0.9.3 `

**Additional context**
I'm using `v0.9.3`

**Comment by VaibhavPage** [comment]
Thanks for reporting the bug. I'll put the fix asap

**Comment by VaibhavPage** [comment]
Fixed in v0.10
